### PR TITLE
linked time: Add min, max, and percentage change columns

### PR DESCRIPTION
**Pull request by JamesHollyer**
* Motivation for features / changes
When we launch linked time we will be allowing a range selection. The data table needs to be able to handle range values when in that mode. This change adds the Min, Max, and Percentage Change columns so the user knows which step is selected for each run.

* Technical description of changes
There are two main changes between these columns and others that have been added. The first is that the min and max have some calculation that I included in a utility function that I put directly in the ScalarCardDataTable. The second is that the percentage change uses arrows similar to the value change column. For code reuse I created an ng-template for the arrow and used that template for both columns.

* Screenshots of UI changes
This is how the UI looks after this PR. In a future PR I will replace the word "Change" with a delta symbol and I plan to make a few other header clean up changes.

Screen Shot 2022-08-17 at 3 06 53 PM

### CODE REVIEW

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts [hunk 1]
```diff
@@ -428,7 +428,10 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
           headers.push(ColumnHeaders.RELATIVE_TIME);
         } else {
           headers.push(ColumnHeaders.RUN);
+          headers.push(ColumnHeaders.MIN_VALUE);
```
Comment (bmd3k) [inline tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts:431]: I would be nice to have a comment saying this else block is for range selection.

File: tensorboard/webapp/widgets/data_table/data_table_component.ng.html [hunk 2]
```diff
@@ -33,22 +33,33 @@
...
+                <ng-container
+                  *ngTemplateOutlet="arrow; context: {$implicit:runData.PERCENTAGE_CHANGE}"
+                ></ng-container>
                 {{ getFormattedDataForColumn(header, runData) }}
               </span>
               <span *ngSwitchDefault
                 >{{ getFormattedDataForColumn(header, runData) }}</span
               >
             </td>
+            <ng-template #arrow let-value>
```
Comment (bmd3k) [inline tensorboard/webapp/widgets/data_table/data_table_component.ng.html:51]: This seems like a slightly odd place to locate your template: Can it be moved to the top-level or closer to the top-level of the dom hierarchy in this file? Maybe just move it to the end of the file?

File: tensorboard/webapp/widgets/data_table/data_table_component.ts [hunk 3]
```diff
@@ -144,6 +150,29 @@ export class DataTableComponent {
         return intlNumberFormatter.formatShort(
           selectedStepRunData.END_VALUE as number
         );
+      case ColumnHeaders.MIN_VALUE:
+        if (selectedStepRunData.MIN_VALUE === undefined) {
+          return '';
+        }
+        return intlNumberFormatter.formatShort(
+          selectedStepRunData.MIN_VALUE as number
```
Comment (bmd3k) [inline tensorboard/webapp/widgets/data_table/data_table_component.ts:158]: slightly odd that you have to cast this 'as number' given that you just checked whether it is undefined. What does the compiler think is the type of selectedStepRunData.MIN_VALUE?

(I guess comment applies multiple places in this switch statement)

**Review by bmd3k — APPROVED** [review]

File: tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts (hunk 1 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts:432]: Done.

File: tensorboard/webapp/widgets/data_table/data_table_component.ng.html (hunk 2 above)
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ng.html]: I thought it was convenient to have a it close to where it is being used. I moved it to the bottom of the file though.

File: tensorboard/webapp/widgets/data_table/data_table_component.ts (hunk 3 above)
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ts:158]: The type is number | string | undefined. So at this point the compiler thinks it is a number | string.

**Review by JamesHollyer — COMMENTED** [review]
Thanks Brian.

**Pull request merged**